main.go

```diff
@@ -47,4 +46,9 @@ func main() {
 	if err = http.ListenAndServe(":8000", r); err != nil {
 		log.Fatal(err)
 	}
+
+	http.Handle("/metrics", promhttp.Handler())
+	if err = http.ListenAndServe(":8001", nil); err != nil {
+		log.Fatal(err)
+	}
```

Comment:
To run "/metrics" on another endpoint we need to initialize a new router and add handler function in that. And also I think `http.ListenAndServe` function is blocking so we need to run first instance as go routine.